runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactStagingServiceTest.java

```diff
@@ -0,0 +1,273 @@
+
+  public static final Joiner JOINER = Joiner.on("");
+  public static final Charset CHARSET = StandardCharsets.UTF_8;
+  private GrpcFnServer<BeamFileSystemArtifactStagingService> server;
+  private BeamFileSystemArtifactStagingService artifactStagingService;
+  private ArtifactStagingServiceStub stub;
+  private Path srcDir;
+  private Path destDir;
+
+  @Before
+  public void setUp() throws Exception {
+    artifactStagingService = new BeamFileSystemArtifactStagingService();
+    server = GrpcFnServer
+        .allocatePortAndCreateFor(artifactStagingService, InProcessServerFactory.create());
+    stub =
+        ArtifactStagingServiceGrpc.newStub(
+            InProcessChannelBuilder.forName(server.getApiServiceDescriptor().getUrl()).build());
+
+    srcDir = Files.createTempDirectory("BFSTemp");
+    destDir = Files.createTempDirectory("BFDTemp");
+
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (server != null) {
+      server.close();
+    }
+    if (artifactStagingService != null) {
+      artifactStagingService.close();
+    }
+    deleteDir(srcDir, "BFSTemp");
+    deleteDir(destDir, "BFDTemp");
+    server = null;
+    artifactStagingService = null;
+    stub = null;
+  }
+
+  private void deleteDir(Path dir, String sanityString) throws IOException {
+    if (dir != null && dir.toAbsolutePath().toString().contains(sanityString)) {
+      Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+          Files.deleteIfExists(file);
+          return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+          Files.deleteIfExists(dir);
+          return FileVisitResult.CONTINUE;
+        }
+      });
+    }
+  }
+
+  private void putArtifact(String stagingSessionToken, final String filePath, final String fileName)
+      throws Exception {
+    StreamObserver<PutArtifactRequest> outputStreamObserver = stub
+        .putArtifact(new StreamObserver<PutArtifactResponse>() {
+          @Override
+          public void onNext(PutArtifactResponse putArtifactResponse) {
+            Assert.fail("OnNext should never be called.");
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            throwable.printStackTrace();
+            Assert.fail("OnError should never be called.");
+          }
+
+          @Override
+          public void onCompleted() {
+          }
+        });
+    outputStreamObserver.onNext(PutArtifactRequest.newBuilder().setMetadata(
+        PutArtifactMetadata.newBuilder()
+            .setMetadata(ArtifactMetadata.newBuilder().setName(fileName).build())
+            .setStagingSessionToken(stagingSessionToken)).build());
+
+    FileInputStream fileInputStream = new FileInputStream(filePath);
+    byte[] buffer = new byte[1 << 20];
```

Comment:
Make 1 << 20 a constant. You could probably make this much smaller, rather than creating multi-mb files just for testing, as what we really want to know is whether multi-chunk uploads work well.